tests/unit/test_multipan_rcp.cpp

```diff
@@ -0,0 +1,795 @@
+        SuccessOrQuit(mEncoder.WriteUint32(sDefaultFrame.mInfo.mTxInfo.mTxDelayBaseTime));
+
+        endFrame("Transmit Frame");
+    }
+
+    void createScanChannelMaskFrame(uint8_t aMask)
+    {
+        startFrame(SPINEL_CMD_PROP_VALUE_SET, SPINEL_PROP_MAC_SCAN_MASK);
+        SuccessOrQuit(mEncoder.WriteUint8(aMask));
+        endFrame("Channel Mask Frame");
+    }
+
+    void createMacScanFrame()
+    {
+        uint8_t state = SPINEL_SCAN_STATE_ENERGY;
+
+        startFrame(SPINEL_CMD_PROP_VALUE_SET, SPINEL_PROP_MAC_SCAN_STATE);
+        SuccessOrQuit(mEncoder.WriteUint8(state));
+        endFrame("Scan State Frame");
+    }
+
+    void createReadStatusFrame()
+    {
+        startFrame(SPINEL_CMD_PROP_VALUE_GET, SPINEL_PROP_LAST_STATUS);
+        endFrame("Read Status Frame");
+    }
+
+    void enableRawLink()
+    {
+        static const bool isLinkEnabled = true;
+        createLinkEnableFrame(isLinkEnabled);
+        sendToRcp();
+    }
+
+    void disableRawLink()
+    {
+        static const bool isLinkEnabled = false;
+        createLinkEnableFrame(isLinkEnabled);
+        sendToRcp();
+    }
+
+    spinel_status_t startTransmit()
+    {
+        static const bool getResponse = true;
+        createTransmitFrame();
+        sendToRcp(getResponse);
+        return static_cast<spinel_status_t>(mNcp->getSpinelStatus());
+    };
+
+    void setScanChannelMask(uint32_t aMask)
+    {
+        createScanChannelMaskFrame(aMask);
+        sendToRcp();
+    }
+
+    uint32_t startEnergyScan()
+    {
+        static const bool getResponse = true;
+        createMacScanFrame();
+        sendToRcp(getResponse);
+        return mNcp->getSpinelStatus();
+    }
+
+    void getCommandStatus()
+    {
+        createReadStatusFrame();
+        sendToRcp();
+    }
+
+    void finishTransmit()
+    {
+        /* Reset instance submac state to sleep by resetting link
+           This is needed for a second transmit command to succeed
+           as the HandleTimer method will not be called to reset the submac */
+        disableRawLink();
+        enableRawLink();
+
+        /* Proceed with transmit done callback from ncp */
+        mNcp->processTransmit();
+    };
+
+private:
+    void startFrame(unsigned int aCommand, spinel_prop_key_t aKey)
+    {
+        uint8_t spinelHeader = SPINEL_HEADER_FLAG | mIid | mTid;
+
+        SuccessOrQuit(mEncoder.BeginFrame(Spinel::Buffer::kPriorityLow));
+        SuccessOrQuit(mEncoder.WriteUint8(spinelHeader));
+        SuccessOrQuit(mEncoder.WriteUintPacked(aCommand));
+        SuccessOrQuit(mEncoder.WriteUintPacked(aKey));
+    }
+
+    void endFrame(const char *aTextMessage)
+    {
+        static const uint16_t display_length = 64;
+        SuccessOrQuit(mEncoder.EndFrame());
+        // DumpBuffer(aTextMessage, mBuf, display_length);
+    }
+
+    void sendToRcp(bool needsResponse = false)
+    {
+        static const uint8_t data_offset     = 2;
```

Comment:
Curious, what do we have at 0 and 1 offset?

Reply:
It looks like another spinel prefix unrelated to the frame itself.
NcpBase subclasses have interface specific offsets within their frames to point to this section of data, and in this case, 2 is the magic frame offset.